File: src/main/java/com/server/ordering/domain/RepresentativeMenu.java
package com.server.ordering.domain;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;

import static javax.persistence.FetchType.*;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.PROTECTED;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
@RequiredArgsConstructor
public class RepresentativeMenu {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "representative_menu_id")
    private Long id;

    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "food_id")
    private Food food;

    @NonNull
    private String foodName;

    public void registerRestaurant(Restaurant restaurant) {
        this.restaurant = restaurant;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/RepresentativeMenu.kt
package com.byeoru.ordering_server.domain

import javax.persistence.*

@Entity
class RepresentativeMenu(restaurant: Restaurant,
                         food: Food,
                         foodName: String) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "representative_menu_id")
     var id: Long? = null
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
     var restaurant: Restaurant = restaurant
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "food_id")
     var food: Food = food
         protected set

     var foodName: String = foodName
         protected set

    fun registerRestaurant(restaurant: Restaurant) {
        this.restaurant = restaurant
    }
}